src/crypto/public_key/elliptic/eddsa.js

```diff
@@ -77,13 +77,15 @@ export async function generate(algo) {
  * @param {Uint8Array} publicKey - Public key
  * @param {Uint8Array} privateKey - Private key used to sign the message
  * @param {Uint8Array} hashed - The hashed message
+ * @param {Boolean} disableHashLengthCheck - whether to skip the hash digest check; this is needed when
+ *  the function is called in the context of PQC composite signatures, that have different hash size requirements.
  * @returns {Promise<{
  *   RS: Uint8Array
  * }>} Signature of the message
  * @async
  */
-export async function sign(algo, hashAlgo, message, publicKey, privateKey, hashed) {
-  if (getHashByteLength(hashAlgo) < getHashByteLength(getPreferredHashAlgo(algo))) {
+export async function sign(algo, hashAlgo, message, publicKey, privateKey, hashed, disableHashLengthCheck = false) {
+  if (!disableHashLengthCheck && getHashByteLength(hashAlgo) < getHashByteLength(getPreferredHashAlgo(algo))) {
```

Comment:
Rather than disabling this check here, I think it'd be cleaner to move this check to the Ed25519 and Ed448 algorithms. That way, the code might also become simpler, e.g. for Ed25519 you could just do
```js
  if (getHashByteLength(hashAlgo) < 32) {
```
exactly as the spec says, rather than calling out to `getPreferredHashAlgo`.

Reply:
The code is currently shared between ed25519 and 448, and with the WebCrypto ed448 support in sight I think there is going to be almost 100% code overlap between them in the future; perhaps here we could take the min hash size as input (defaulting to `getHashByteLength(getPreferredHashAlgo(algo))` instead of the flag to disable the check.
But yeah let's delay this topic until upstream integration, also considering there is another upcoming PR there for WebCrypto integration that affects the eddsa code as well 🙂

Reply:
No, my idea was to move the check to the algorithms [here](https://github.com/openpgpjs/openpgpjs/blob/2a8969b4370a7f73792faa4ac67d2e4bc2ec474d/src/crypto/signature.js#L173) and [here](https://github.com/openpgpjs/openpgpjs/blob/2a8969b4370a7f73792faa4ac67d2e4bc2ec474d/src/crypto/signature.js#L119) (I see now that the code is shared also there so there's not even a need to duplicate it). That way, it doesn't affect PQC+EdDSA.